### PR TITLE
Fix corrupt database when saving is interrupted

### PR DESCRIPTION
**Pull request by omfgicbf**
This fixes the order of writes and adds appropriate flushes when committing to disk.

This ensures that the journal is written to disk before proceeding with any changes and avoids leaving the database in an inconsistent state in case of a power failure.

### CODE REVIEW

File: LiteDB/Engine/Services/TransactionService.cs [hunk 1]
```diff
@@ -75,17 +75,32 @@ public void PersistDirtyPages()
 
                 // mark header as recovery before start writing (in journal, must keep recovery = false)
                 header.Recovery = true;
+
+                // flush to disk to ensure journal is committed to disk before proceeding
+                _disk.Flush();
```
Comment (mbdavid) [inline LiteDB/Engine/Services/TransactionService.cs]: `WriteJournal` method contains a a `Flush` call at end:

https://github.com/mbdavid/LiteDB/blob/v4/LiteDB/Engine/Disks/FileDiskService.cs#L194

Comment (omfgicbf) [reply LiteDB/Engine/Services/TransactionService.cs]: Yes, but `header.Recovery = true` is lost if the write is interrupted after that line. As there is no guarantee to the order of writes, if the changes below are committed but this is not, then you end up with a corrupt database.

Comment (mbdavid) [reply LiteDB/Engine/Services/TransactionService.cs]: You right, I miss this. I will accept your PR, thanks. New 4.1.5 will be released soon.

Comment (AviadKatani) [reply LiteDB/Engine/Services/TransactionService.cs:80]: Hi, if I already have a corrupted database from power failure, how can I fix it?

**Pull request merged**